projects/packages/videopress/src/client/state/utils/map-videos.ts

```diff
@@ -12,3 +19,52 @@ export const mapVideo = ( video: OriginalVideoPressVideo ): VideoPressVideo => {
 export const mapVideos = ( videos: OriginalVideoPressVideo[] ): VideoPressVideo[] => {
 	return videos.map( mapVideo );
 };
+
+export const mapVideoFromWPV2MediaEndpoint = (
+	video: OriginalVideoPressVideo
+): VideoPressVideo => {
+	const { media_details: mediaDetails, id, caption } = video;
+
+	const { videopress: videoPressMediaDetails, width, height } = mediaDetails;
+
+	const {
+		title,
+		description,
+		original: url,
+		poster,
+		upload_date: date,
+		duration,
+		is_private: isPrivate,
+		file_url_base: fileURLBase,
+		files,
+	} = videoPressMediaDetails;
+
+	const { dvd } = files;
+
+	// Pick poster image from dvd file type.
+	const dvdImage = `${ fileURLBase.https }${ dvd.original_img }`;
+
+	return {
```

Comment:
Can you return `guid` too? It will be used

Reply:
yes. just FYI the response is based on what the types defines. Let's update both.

Reply:
yep, I updated the type at #26281, but is much better do it here